server/controllers/payroll/reports/payslipGenerator.js

```diff
@@ -0,0 +1,234 @@
+    footerFontSize : '7',
+  });
+
+  let report;
+  const getPaiementRubrics = [];
+  const getHolidays = [];
+  const getOffDays = [];
+  const employeeData = [];
+
+  const data = {};
+
+  data.enterprise = req.session.enterprise;
+  data.user = req.session.user;
+  data.lang = options.lang;
+
+  // set up the report with report manager
+  try {
+    report = new ReportManager(TEMPLATE, req.session, options);
+  } catch (e) {
+    next(e);
+    return;
+  }
+
+  options.employees.forEach(emp => {
+    /**
+    * If the options.employees content is an Array (When selecting multiple employees)  
+    * each element of the array must be converted to Json
+    */
+    const employee = typeVar ? JSON.parse(emp) : emp;
+
+    employeeData.push({ employee });
+
+    const sql = `
+      SELECT rubric_paiement.paiement_uuid, rubric_paiement.value AS result, 
+      BUID(paiement.employee_uuid) AS employee_uuid, rubric_payroll.abbr, rubric_payroll.label, 
+      rubric_payroll.is_percent, rubric_payroll.value, rubric_payroll.is_discount, 
+      rubric_payroll.is_social_care, rubric_payroll.is_employee
+      FROM rubric_paiement
+      JOIN paiement ON paiement.uuid = rubric_paiement.paiement_uuid
+      JOIN rubric_payroll ON rubric_payroll.id = rubric_paiement.rubric_payroll_id
+      WHERE paiement.payroll_configuration_id = ? AND paiement.employee_uuid = ?
+      ORDER BY rubric_payroll.label, rubric_payroll.is_social_care ASC, rubric_payroll.is_discount ASC 
+    `;
+
+    const sqlHolidayPaiement = `
+      SELECT holiday_paiement.holiday_nbdays, holiday_paiement.holiday_nbdays, holiday_paiement.holiday_percentage, 
+      holiday_paiement.label, holiday_paiement.value, BUID(holiday_paiement.paiement_uuid) AS paiement_uuid
+      FROM holiday_paiement
+      WHERE holiday_paiement.paiement_uuid = ?
+    `;
+
+    const sqlOffDayPaiement = `
+      SELECT offday_paiement.offday_percentage, BUID(offday_paiement.paiement_uuid) AS paiement_uuid,
+      offday_paiement.label, offday_paiement.value
+      FROM offday_paiement
+      WHERE offday_paiement.paiement_uuid = ?
+    `;
+
+    getPaiementRubrics.push({
+      query : sql,
+      params : [options.idPeriod, db.bid(employee.employee_uuid)],
+    });
+
+    getHolidays.push({
+      query : sqlHolidayPaiement,
+      params : [db.bid(employee.uuid)],
+    });
+
+    getOffDays.push({
+      query : sqlOffDayPaiement,
+      params : [db.bid(employee.uuid)],
+    });
+  });
+
+  // initialise the transaction handler
+  const transaction = db.transaction();
+  const transactionHoliday = db.transaction();
+  const transactionOffDay = db.transaction();
+
+  getPaiementRubrics.forEach(item => {
+    transaction.addQuery(item.query, item.params);
+  });
+
+  getHolidays.forEach(item => {
+    transactionHoliday.addQuery(item.query, item.params);
+  });
+
+  getOffDays.forEach(item => {
+    transactionOffDay.addQuery(item.query, item.params);
+  });
+
+  transaction.execute()
```

Comment:
If you are just using SELECTs a transaction will be a lot slower.  Go ahead and use an array of promises with `q.all()`.